### PR TITLE
p2p/enode: add quic ENR entry

### PR DESCRIPTION
**Pull request by guillaumemichel**
Add `quic` entry to the ENR as proposed in https://github.com/ethereum/consensus-specs/pull/3644

Allowing to read [Lighthouse `quic` port](https://github.com/sigp/lighthouse/blob/d6ba8c397557f5c977b70f0d822a9228e98ca214/book/src/developers.md?plain=1#L20-L26) from ENR.

### CODE REVIEW

File: p2p/enode/node.go [hunk 1]
```diff
@@ -184,6 +189,11 @@ func (n *Node) TCP() int {
 	return int(n.tcp)
 }
 
+// QUIC returns the QUIC port of the node.
+func (n *Node) QUIC() int {
```
Comment (fjl) [inline p2p/enode/node.go:193]: I think it should return `uint16`.

**Comment by fjl** [comment]
Not sure if we actually need the accessor on `enode.Node` at this point.

File: p2p/enode/node.go (hunk 1 above)
Comment (guillaumemichel) [reply p2p/enode/node.go:193]: I copied the `UDP()` and `TCP()` return types. But I can update them to have `uint16` as return type.

https://github.com/ethereum/go-ethereum/blob/83e70aa3d00bbee9713b31768a30a7741eee8945/p2p/enode/node.go#L177-L185

File: p2p/enode/node.go [hunk 2]
```diff
@@ -200,6 +200,16 @@ func (n *Node) TCPEndpoint() (netip.AddrPort, bool) {
 	return netip.AddrPortFrom(n.ip, n.tcp), true
 }
 
+// QUICEndpoint returns the announced QUIC endpoint.
+func (n *Node) QUICEndpoint() (netip.AddrPort, bool) {
+	var quic enr.QUIC
+	n.Load(&quic)
+	if !n.ip.IsValid() || n.ip.IsUnspecified() || quic == 0 {
```
Comment (fjl) [inline p2p/enode/node.go:207]: Here we need to distinguish if the IP is IPv6, and try loading QUIC6 if it's there.

Comment (guillaumemichel) [reply p2p/enode/node.go:211]: Just addressed it. Let me know if it looks right

**Comment by lightclient** [comment]
I added a few more tests here. It didn't seem like the included test was really verifying the port was loaded correctly. Also verifying we don't return an endpoint with a quic6 port for IPv4 and vice versa. Not sure how important that is.

**Review by lightclient — APPROVED** [review]

**Comment by cortze** [comment]
Any update on the status of PR?

**Comment by guillaumemichel** [comment]
Is the PR good for merge?

**Pull request merged**